### PR TITLE
11 yes no question

### PR DESCRIPTION
**Pull request by yvankamdem**
implementation de l'envoi d'un modèle de message predefini via l'api de facebook

### CODE REVIEW

File: app4java/src/main/java/com/nyxei/pollingus/PollingusApplication.java [hunk 1]
```diff
@@ -1,13 +1,37 @@
...
+import java.net.http.HttpResponse;
+import java.net.http.HttpResponse.BodyHandlers;
+
 import org.springframework.boot.autoconfigure.SpringBootApplication;
 
 @SpringBootApplication
 public class PollingusApplication {
 
 	public static void main(String[] args) {
-		SpringApplication.run(PollingusApplication.class, args);
```
Comment (SanAlexis) [inline app4java/src/main/java/com/nyxei/pollingus/PollingusApplication.java:10]: Main method role is not to launch request but to launch the whole program. You should have a better place/object/class to do this job.

File: app4java/src/main/java/com/nyxei/pollingus/PollingusApplication.java [hunk 2]
```diff
@@ -1,13 +1,37 @@
...
+		 
+
+		try {
+			HttpRequest request = HttpRequest.newBuilder()
+				.uri(new URI("https://graph.facebook.com/v15.0/102062986090216/messages"))
```
Comment (SanAlexis) [inline app4java/src/main/java/com/nyxei/pollingus/PollingusApplication.java:21]: The URI parameter should be a constant called whenever you need it.

File: app4java/src/main/java/com/nyxei/pollingus/PollingusApplication.java [hunk 3]
```diff
@@ -1,13 +1,37 @@
...
+				.header("Authorization", "Bearer EAAGYN0OOscYBAKr9XGGCllZBFZC9XAIJDwAVGAlkDMWdhMElo6aJbWbsMu2W7pZCfObN39uAHqdZAcDNx5qmPlVv88wmiZBva7YnEqzJ3C2hzUiZAFZCdD9zCjUQ193touuz6qqE3VNgwCKtYhW0Cq8bkgxtgdx2COdx5VRwsEcbQjO3w7VQazLgMvy7TxsR9ZANIvIgdFcYTWY1RZAvwDuSH")
```
Comment (SanAlexis) [inline app4java/src/main/java/com/nyxei/pollingus/PollingusApplication.java:22]: Same goes for all the params you use.

File: app4java/src/main/java/com/nyxei/pollingus/PollingusApplication.java [hunk 4]
```diff
@@ -1,13 +1,37 @@
...
 
 @SpringBootApplication
 public class PollingusApplication {
 
 	public static void main(String[] args) {
-		SpringApplication.run(PollingusApplication.class, args);
+		 
+
+		try {
+			HttpRequest request = HttpRequest.newBuilder()
```
Comment (SanAlexis) [inline app4java/src/main/java/com/nyxei/pollingus/PollingusApplication.java:20]: request is really a poor name.

File: app4java/src/main/java/com/nyxei/pollingus/PollingusApplication.java [hunk 5]
```diff
@@ -1,13 +1,37 @@
...
+				.uri(new URI("https://graph.facebook.com/v15.0/102062986090216/messages"))
+				.header("Authorization", "Bearer EAAGYN0OOscYBAKr9XGGCllZBFZC9XAIJDwAVGAlkDMWdhMElo6aJbWbsMu2W7pZCfObN39uAHqdZAcDNx5qmPlVv88wmiZBva7YnEqzJ3C2hzUiZAFZCdD9zCjUQ193touuz6qqE3VNgwCKtYhW0Cq8bkgxtgdx2COdx5VRwsEcbQjO3w7VQazLgMvy7TxsR9ZANIvIgdFcYTWY1RZAvwDuSH")
+				.header("Content-Type", "application/json")
+				 .POST(HttpRequest.BodyPublishers.ofString("{ \"messaging_product\": \"whatsapp\", \"recipient_type\": \"individual\", \"to\": \"237694219581\", \"type\": \"template\", \"template\": { \"name\": \"nouvelle_offre\", \"language\": { \"code\": \"fr\" } } }"))
+				 // .POST(HttpRequest.BodyPublishers.ofString("{ \"messaging_product\": \"whatsapp\", \"recipient_type\": \"individual\", \"to\": \"237694219581\", \"type\": \"template\", \"template\": { \"name\": \"_new_customer_offer\", \"language\": { \"code\": \"fr\" }, \"components\":[{\"type\":\"header\",\"parameters\":[{\"type\":\"image\",\"image\":{\"link\":\"https://images.app.goo.gl/HFdZU5Zf1Sx5wyBb9\"}}]}, ] } }"))
+				.build();
+			HttpClient http = HttpClient.newHttpClient();
```
Comment (SanAlexis) [inline app4java/src/main/java/com/nyxei/pollingus/PollingusApplication.java:27]: http, response are poor names.

**Review by SanAlexis — CHANGES_REQUESTED** [review]
# Good
You achieved to communicate with the Meta API.

# Bad
The purpose of the issue is far to be achieved.
* Where do you consume description file ?
* Where do you consume response and do something about it ?
* Where do you select the receiver of the message ?
* The architecture of the solution is pretty bad.

File: app4java/src/main/java/com/nyxei/pollingus/PollingusApplication.java (hunk 1 above)
Comment (yvankamdem) [reply app4java/src/main/java/com/nyxei/pollingus/PollingusApplication.java:10]: je comprend mais je n'arrive pas à dissocier la requête de la méthode... si tu peux nous donnez une architecture pour ça

File: app4java/src/main/java/com/nyxei/pollingus/PollingusApplication.java (hunk 2 above)
Comment (yvankamdem) [reply app4java/src/main/java/com/nyxei/pollingus/PollingusApplication.java:21]: final String URI_rappel = "https://graph.facebook.com/v15.0/102062986090216/messages"

File: app4java/src/main/java/com/nyxei/pollingus/PollingusApplication.java (hunk 3 above)
Comment (yvankamdem) [reply app4java/src/main/java/com/nyxei/pollingus/PollingusApplication.java:22]: final String Token = "Bearer EAAGYN0OOscYBAJAlwOwszT65lOTGZAFG29ZA3A9DBJZADGEdOCNZChoEm9eE34pWSo6WYWuDOs2pBKCCNGoiqqhBp9uRnjFpU61ZAqlxalJXGJFfcIDZADATBCZBhe2MZATNJzFkjCVZAXsYJ3vwrdjOZCDuNus7ZA6390ZASSK9bJWY04lU0RJpKXeYz2Bd48G232kMy1bzM8Be9QZDZD";
		final String App_json = "application/json"

File: app4java/src/main/java/com/nyxei/pollingus/PollingusApplication.java (hunk 1 above)
Comment (SanAlexis) [reply app4java/src/main/java/com/nyxei/pollingus/PollingusApplication.java:10]: Pas d'avancée ici.

**Review by SanAlexis — CHANGES_REQUESTED** [review]
Toujours pas d'avancée avec cette classe.

File: app4java/src/main/java/com/nyxei/pollingus/PollingusApplication.java (hunk 1 above)
Comment (yvankamdem) [reply app4java/src/main/java/com/nyxei/pollingus/PollingusApplication.java:10]: problème résolu dans le nouveau PR que j'ai effectué

**Review by yvankamdem — COMMENTED** [review]
suppression de la requête dans le main

**Pull request closed without merging**